# ãƒ—ãƒ­ã‚°ãƒ©ãƒãƒ¼ï¼ˆProgrammerï¼‰
extends Node


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# ãƒãƒ¼ãƒ‰ãƒ»ãƒ‘ã‚¹é–¢é€£
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼


# ãƒ‡ã‚£ãƒ¬ã‚¯ã‚¿ãƒ¼ãƒ»ãƒãƒ–å–å¾—
func monkey():
	return $"ğŸµMonkey"


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# ã‚¢ãƒ‰ã‚ªãƒ³ãƒ»ã‚¹ãƒ­ãƒƒãƒˆ
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼


# BGMãƒãƒ¼ãƒ‰ã®ã‚­ãƒ£ãƒƒã‚·ãƒ¥
var bg_musics = null

# ã‚¤ãƒ©ã‚¹ãƒˆãƒ»ãƒãƒ¼ãƒ‰ã®ã‚­ãƒ£ãƒƒã‚·ãƒ¥
var images = null

# ã‚µã‚¦ãƒ³ãƒ‰ãƒ»ã‚¨ãƒ•ã‚§ã‚¯ãƒˆï¼ˆSound Effect, Sound FXï¼›åŠ¹æœéŸ³ï¼‰ã®ã‚­ãƒ£ãƒƒã‚·ãƒ¥
var sound_fx = null

# ãƒ†ãƒ­ãƒƒãƒ—ã®ã‚­ãƒ£ãƒƒã‚·ãƒ¥
var telops = null


# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼
# æº–å‚™
# ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼ãƒ¼


func _ready():
	self.bg_musics = MonkeyHand.create(
			self.monkey().of_director().musician_bg_musics())		# æ¢ã™å ´æ‰€

	self.images = MonkeyHand.create(
			self.monkey().of_director().illustrator())			# æ¢ã™å ´æ‰€

	self.sound_fx = MonkeyHand.create(
			self.monkey().of_director().musician_sound_fx())			# æ¢ã™å ´æ‰€

	self.telops = MonkeyHand.create(
			self.monkey().of_director().telop_coordinator())	# æ¢ã™å ´æ‰€
